config/src/content/jcr_root/apps/core/wcm/config/com.adobe.cq.wcm.core.components.internal.servlets.TableOfContentsFilter.config

```diff
@@ -0,0 +1,15 @@
+# Copyright 2022 Adobe
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+enabled="true"
```

Comment:
shouldn't it be `false`, similar to the default value in `TableOfContentsFilter.java`?
(https://github.com/adobe/aem-core-wcm-components/pull/2247/files#diff-00843a122ef914854877a70c7a9d0edc4d79d2363c66279590703f50f75d4a42R89

Reply:
@vsarya If the configuration is only about enabling or disabling the filter you can also just set the `org.osgi.service.component.annotations.Component#configurationPolicy` to require. If no configuration present the whole component gets keeps disabled.

Reply:
@jckautzmann We want `TableOfContentsFilter` to be enabled when a config is present and be disabled otherwise, as mentioned by @vladbailescu in his comment on [SITES-6894](https://jira.corp.adobe.com/browse/SITES-6894).

> Discussed with [Gabriel Walt](https://jira.corp.adobe.com/secure/ViewProfile.jspa?name=gwalt), [Abhinav Saraswat](https://jira.corp.adobe.com/secure/ViewProfile.jspa?name=absarasw), [Vishal Singh Arya](https://jira.corp.adobe.com/secure/ViewProfile.jspa?name=varya) and [Mirko Sagolj](https://jira.corp.adobe.com/secure/ViewProfile.jspa?name=msagolj). Gabriel insists we keep the rest of the features from 2.20.x (asset delivery) for the June release. We agreed to update the TOC Filter to be enabled only when an OSGi config is present. On AEMaaCS that config will not be present by default. Abhinav and Vishal will be updating the code and testing it on OneWellington upgraded instance, Mirko and I will review and help with the Core Components patch release (2.20.6).

Reply:
@vsarya, you can combine the two so that:
- if the config is not present the `TableOfContentsFilter` si disabled completely
- if the config is present the filter will check the `enabled` property and only do the processing if it's `true`

Reply:
@vsarya Keeping the component disabled has also the advantage that the Filter will not show up in the FilterChain during request processing.

Reply:
@bpauli @vladbailescu I have done the required changes.